devtools_m1/php_utils/create_product.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+require_once __DIR__.'/../abstract.php';
+
+class Mage_Shell_Drip_CreateProduct extends Mage_Shell_Abstract
+{
+    public function run()
+    {
+        $stdin = fopen('php://stdin', 'r');
+        $data = stream_get_contents($stdin);
+        $json = json_decode($data, true);
+
+        if ($json === null) {
+            throw new \Exception('Null JSON parse');
+        }
+
+        $type = $json['typeId'];
+        switch ($type) {
+            case 'simple':
+            case '':
+            case null:
+                $this->buildSimpleProduct($json)->save();
+                break;
+            case 'configurable':
+                $this->buildConfigurableProduct($json);
+                break;
+            case 'grouped':
+                $this->buildGroupedProduct($json);
+                break;
+            case 'bundle':
+                $this->buildBundleProduct($json);
+                break;
+            default:
+                throw new \Exception("Unsupported type: ${type}");
+        }
+    }
+
+    protected function buildSimpleProduct($data)
+    {
+        $product = Mage::getModel('catalog/product');
+
+        $defaultAttrSetId = Mage::getModel('catalog/product')->getDefaultAttributeSetId();
+
+        $defaults = array(
+            "storeId" => 1,
+            "websiteIds" => [1],
+            "typeId" => "simple",
+            "weight" => 4.0000,
+            "status" => 1, //product status (1 - enabled, 2 - disabled)
+            "taxClassId" => 0, //tax class (0 - none, 1 - default, 2 - taxable, 4 - shipping)
+            "price" => 11.22,
+            "cost" => 22.33,
+            "attributeSetId" => $defaultAttrSetId,
+            "createdAt" => strtotime('now'),
+            "updatedAt" => strtotime('now'),
+            "stockData" => array(
+                "use_config_manage_stock" => 0,
+                "manage_stock" => 1,
+                "is_in_stock" => 1,
+                "qty" => 999
+            ),
+        );
+        $fullData = array_replace_recursive($defaults, $data);
+
+        // This assumes that you properly name all of the attributes. But we control both ends, so it should be fine.
+        foreach ($fullData as $key => $value) {
+            $methodName = "set".ucfirst($key);
```

Comment:
This isn't... terrible... but could you have a method on the `catalog/product` class like `updateValuesFromArray`?

Reply:
You absolutely could, but given that I control the JSON coming in, I decided not to be too picky about it. This is only used by tests.

Reply:
I'm sorry, I just re-read this. I'm not sure whether that mechanism exists, I'll look.

Reply:
Since this is controlled on both ends, I'm going to leave it as is for now. It's noted as an improvement later.